File: 351001/Strizhak/rest_noSql/Publisher/Entities/Topic.cs
using Publisher.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Publisher.Entities
{
    public class Topic: IEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        public long UserId { get; set; } 

        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Title { get; set; } = null!;

        [Required]
        [StringLength(2048, MinimumLength = 4)]
        public string Content { get; set; } = null!;

        [Required]
        public DateTime Created { get; set; } 

        [Required]
        public DateTime Modified { get; set; }

        // Навигация
        public User User { get; set; } = null!;
        public ICollection<TopicTag> TopicTags { get; set; } = new List<TopicTag>();
    }
}

File: 351004/Lavrenov/Lab_1/models/topic.py
from dataclasses import dataclass, field
from typing import List, Optional
from datetime import datetime
from .base import BaseEntity


@dataclass
class Topic(BaseEntity):
    userId: int
    title: str
    content: str
    markerIds: List[int] = field(default_factory=list)
    id: int = None
    created: Optional[datetime] = None
    modified: Optional[datetime] = None
